src/gui/qgsexpressionbuilderwidget.cpp

```diff
@@ -1320,6 +1356,302 @@ void QgsExpressionBuilderWidget::removeSelectedUserExpression()
+  {
+    // validate the type of the array element, can be anything
+    if ( ! expressionValue.isObject() )
+    {
+      // try to stringify and put and indicator what happened
+      skippedExpressionLabels.append( expressionValue.toString() );
+      continue;
+    }
+
+    QJsonObject expressionObj = expressionValue.toObject();
+
+    // make sure the required keys are the correct types
+    if ( ! expressionObj["name"].isString()
+         || ! expressionObj["type"].isString()
+         || ! expressionObj["expression"].isString()
+         || ! expressionObj["group"].isString()
+         || ! expressionObj["description"].isString() )
+    {
+      // try to stringify and put an indicator what happened. Try to stringify the name, if fails, go with the expression.
+      if ( ! expressionObj["name"].toString().isEmpty() )
+        skippedExpressionLabels.append( expressionObj["name"].toString() );
+      else
+        skippedExpressionLabels.append( expressionObj["expression"].toString() );
+
+      continue;
+    }
+
+    // we want to import only items of type expression for now
+    if ( expressionObj["type"].toString() != "expression" )
+    {
+      skippedExpressionLabels.append( expressionObj["name"].toString() );
+      continue;
+    }
+
+    // we want to import only items of type expression for now
+    if ( expressionObj["group"].toString() != "user" )
+    {
+      skippedExpressionLabels.append( expressionObj["name"].toString() );
+      continue;
+    }
+
+    const QString label = expressionObj["name"].toString();
+    const QString expression = expressionObj["expression"].toString();
+    const QString helpText = expressionObj["description"].toString();
+
+    // make sure they have valid name
+    if ( label.contains( "\\" ) || label.contains( "/" ) )
+    {
+      skippedExpressionLabels.append( expressionObj["name"].toString() );
+      continue;
+    }
+
+    settings.beginGroup( label );
+    const QString oldExpression = settings.value( QStringLiteral( "expression" ) ).toString();
+    settings.endGroup();
+
+    // TODO would be nice to skip the cases when labels and expressions match
+    if ( mUserExpressionLabels.contains( label ) && expression != oldExpression )
+    {
+      if ( ! isApplyToAll )
+        showMessageBoxConfirmExpressionOverwrite( isApplyToAll, isOkToOverwrite, label, oldExpression, expression );
+
+      if ( isOkToOverwrite )
+        saveToUserExpressions( label, expression, helpText );
+      else
+      {
+        skippedExpressionLabels.append( label );
+        continue;
+      }
+    }
+    else
+    {
+      saveToUserExpressions( label, expression, helpText );
+    }
+  }
+
+  loadUserExpressions( );
+
+  if ( ! skippedExpressionLabels.isEmpty() )
+  {
+    QStringList skippedExpressionLabelsQuoted;
+    for ( const QString &skippedExpressionLabel : skippedExpressionLabels )
+      skippedExpressionLabelsQuoted.append( QString( "'%1'" ).arg( skippedExpressionLabel ) );
```

Comment:
```suggestion
      skippedExpressionLabelsQuoted.append( QStringLiteral( "'%1'" ).arg( skippedExpressionLabel ) );
```

same for other `QString("...")` except when it's an empty string, `QString()` is preferred in that case.